web_session_allow_public/models/res_users.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 LasLabs Inc.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import models
+
+
+class ResUsers(models.Model):
+
+    _inherit = 'res.users'
+
+    def check(self, db, uid, passwd):
+        """ Catch AccessDenied error and allow public session to proceed """
+        if not passwd:
+            cr = self.pool.cursor()
+            try:
+                is_public = self._has_group(cr, uid, 'base.group_public')
```

Comment:
Admittedly I didn't try it. I'm pretty sure I am hitting the non-api method here though because I'm not within an api method & don't have an env yet. The core code has three methods, api.v7 api.v8 and no api.